docs/APP-SETUP.md

```diff
@@ -0,0 +1,216 @@
+# RHDH Setup Guide For GitHub and GitLab
+
+## Table of Contents
+
+- [GitHub Personal Access Token (PAT)](#github-personal-access-token-pat)
+  - [Prerequisites](#prerequisites)
+  - [Procedure](#procedure)
+- [GitHub App](#github-app)
+  - [Procedure](#procedure-1)
+  - [App Permissions](#app-permissions)
+- [GitLab Personal Access Token (PAT)](#gitlab-personal-access-token-pat)
+  - [Prerequisites](#prerequisites-1)
+  - [Procedure](#procedure-2)
+- [GitLab App](#gitlab-app)
+  - [Procedure](#procedure-3)
+- [Quay Setup](#quay-setup)
+  - [Procedure](#procedure-4)
+- [Homepage / Callback / Webhook URLs](#homepage--callback--webhook-urls)
+  - [Homepage](#homepage-url)
+  - [Callback](#callback-url)
+  - [Webhook](#webhook-url)
+- [Troubleshooting](#troubleshooting)
+  - [Git Resource Not Accessible](#git-resource-not-accessible)
+  - [Pipeline Webhook Not Working](#pipeline-webhook-not-working)
+  - [Pipeline Failed To Push/Pull Image From Registry](#pipeline-failed-to-pushpull-image-from-registry)
+  - [Pipelines Fail To Run For Private Repositories](#pipelines-fail-to-run-for-private-repositories)
+
+## GitHub Personal Access Token (PAT)
+
+### Prerequisites
+
+- A GitHub account.
+
+### Procedure
+
+1. See [creating a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic) on the GitHub official documentation.
+2. For the scopes you should select `repo`, selecting this option will include all scopes within `repo`.
+
+![Example of Repo Scope](../assets/repo-scope-example.png)
+
+## GitHub App
+
+### Procedure
+
+1. Create a GitHub Organization
+2. Follow [Registering a GitHub App](https://docs.github.com/en/apps/creating-github-apps/registering-a-github-app/registering-a-github-app) to create the base of your GitHub App. You can use the default settings unless instructed otherwise.
+   1. For `Homepage URL`, `Webhook URL` and `Callback URL` you can put any value as a placeholder as we will deal with these at a later step.
+   2. Deselect `Expire user authorization tokens`.
+   3. You will need to create a `webhook secret` as part of the App setup process.
+3. You should reference [App Permissions](#app-permissions) to set the proper permissions for your GitHub App. This step is crucial in ensuring functionality.
+4. After app creation, at the top of the page you will be prompted to create a `private key`. Do this. You also need to generate a `client secret`.
+5. In the `Developer Settings` of your newly created GitHub App, navigate to `Install App` and install it to your Organization.
+
+**Note:** It is helpful as part of this step to store the following information somewhere safe:
+- GitHub App ID
+- Client ID
+- Client secret
+- Webhook secret
+- Private key
+
+### App Permissions
```

Comment:
are the followings minimal requirements?

I'm wondering why events like `Issue comment` etc need to be subscribed?

Reply:
good point

Reply:
Let me double check, I may have only checked minimums for the permissions and not subscriptions since we may be able to eliminate the `Issue comment` and `commit comment` subs

Reply:
+1 on the point, btw I thought this was included in the scope of [DEVAI-178](https://issues.redhat.com/browse/DEVAI-178) where the content was decided.

Reply:
AFAIR, only pull and push events needs to be subscribed. 
and under `Organization Permissions` only `members` is needed

also for the repository permission, I saw the list in the PR is having more than what listed in the backstage doc: https://backstage.io/docs/integrations/github/github-apps/#app-permissions

might want to revisit to see if all the permissions listed are required. 
we only want to have the minimum requirement listed here. 
for example, I don't know if we need those
 ```
 -**Checks** - Read & Write
- **Commit Statuses** - Read
```

Reply:
I will revisit the permissions but from my testing on trying to get the minimums and referencing the docs this is what is required - As for the subscriptions @thepetk and I discovered that without the dispatches the PR auto-merge and dispatch doesn't work